### ISSUE TITLE
Missing audio channel information for DTS:X streams?

### ISSUE THREAD

**Issue by rfgamaral** [MediaArea/MediaInfo#299]
Using the "DTS:X Out of the Box (Short)" sample file that can be downloaded from [here](https://thedigitaltheater.com/index.php/dts-trailers/), MediaInfo reports the following channel information for audio track 1:

```
Channel(s)                       : 8
Channel(s)/String                : 8 channel3
Channel(s)_Original              : Object Based
Channel(s)_Original/String       : Object Based
ChannelPositions                 : Object Based
ChannelPositions/String2         : Object Based
ChannelLayout                    : Object Based
```

From this information alone is usually enough to parse the number of channels of any stream, especially from the `ChannelsPositions/String2` parameter, but not on this particular case. This is a `7.1` channel stream but there's no way to know that from media info parameters.

Not sure if this is a bug, an oversight or a feature missing or maybe this is how it's supposed to be? If so, should we just infer it's a 7.1 channel stream because `Channel(s)` reports `8` and it's a DTS:X stream? Or is there a better way?

**Comment by JeromeMartinez** [comment]
This is unexpected behavior from a big change I have made on how to represent Dolby Atmos, and I did not check enough how it is displayed with DTS:X.
I'll show 7.1 in "normal" channels + the "X" part in another line as I do now with Dolby Atmos, for more coherency.

**Comment by rfgamaral** [comment]
Awesome, thanks :)

**Comment by rfgamaral** [comment]
> I'll show 7.1 in "normal" channels + the "X" part in another line as I do now with Dolby Atmos, for more coherency.

Thinking about this, I'm a little confused...

As you previously mentioned to me before [here](https://github.com/MediaArea/MediaInfo/issues/286), `Format_AdditionalFeatures` already contains `XLL X` which identifies a DTS:X track, no need to add anything extra.

I think what's missing from DTS:X tracks is what it's already shown on `ChannelPositions/String2` for Atmos tracks, something like `3/2/2.1` (which gives you `7.1` if you sum up).

Does it make sense or am I misunderstanding what you said?

**Comment by JeromeMartinez** [comment]
You're right. I would show the count of objects in DTS:X when I implement the parsing (IIRC I found no spec about it so for the moment I don't show the count of objects)

**Comment by CarbonFactory** [comment]
Any idea when this will be implemented ? 
I was depending on ChannelPositions/String2 to sum and arrive at the Channels value.

**Comment by JeromeMartinez** [comment]
> Any idea when this will be implemented ?

As indicated in the previous message the main issue is still the lack of spec (or more details about how to detect it), so if you find it, when it will be implemented is likely (depending on the complexity) when you (or someone else) find a spec + few days.

**Comment by JeromeMartinez** [comment]
Wait, I may have misunderstood the question, do you speak about DTX:X itself or the regression after some changes? For the regression, it is when I have some free time, after family time and [paid support](https://mediaarea.net/Services) (they also complain a bit about my delays). I don't like this regression, but some other things are more important for me for the moment.

**Comment by CarbonFactory** [comment]
@JeromeMartinez I meant the `count of objects` like `3/2/2.1`.
As of now, I'm using ffprobe to get the channels as a workaround. It gives 7.1 as output :sweat_smile:

I understand the lack of spec is a roadblock. May be I'll try to see how FFProbe is identifying it, whenever time permits.

Since DTS:X is not used as much as Atmos, you can take as much time as you need :wink: 
Thanks !

**Comment by JeromeMartinez** [comment]
> I meant the count of objects like 3/2/2.1.

This one is a regression, so should be easy to fix. I try to handle that this week.

IIRC, FFmpeg does not catch DTS:X too.

**Comment by rg9400** [comment]
Hello Jerome, by any chance, was the count of objects for DTS X fixed in version 20.03? It would help me decide if it makes sense to update my version.

**Comment by JeromeMartinez** [comment]
Nothing was done in DTS:X in this version (I got lot of work with the AC-4 and MPEG-H stuff and they were the priority, I'll check this issue after a couple of other priorities shorter to develop).

**Comment by Silent-Remux** [comment]
Hi @JeromeMartinez was there any movement on this?

**Comment by rg9400** [comment]
Hi @JeromeMartinez 

Anything further on this? I am mainly interested in a `count of objects` like `3/2/2.1`, which you mentioned should be easier to fix compared to the whole issue

**Comment by ValZapod** [comment]
Does DTS Express have DTS:X metadata too? Like EAC3? What about DTS next generation Ultra?
